pallets/ajuna-affiliate/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+			extrinsic_id: ExtrinsicId,
+		},
+		RuleExecuted {
+			extrinsic_id: ExtrinsicId,
+			account: T::AccountId,
+			beneficiary: T::AccountId,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// There is no account set as the organizer
+		OrganizerNotSet,
+		/// An account cannot affiliate itself
+		CannotAffiliateSelf,
+		/// The account is not allowed to receive affiliates
+		CannotAffiliateToAccount,
+		/// This account has reached the affiliate limit
+		CannotAffiliateMoreAccounts,
+		/// This account has already been affiliated by another affiliator
+		CannotAffiliateAlreadyAffiliatedAccount,
+		/// This account is already an affiliator, so it cannot be affiliated by another account
+		CannotAffiliateExistingAffiliator,
+		/// The given extrinsic identifier is already paired with an affiliate rule
+		ExtrinsicAlreadyHasRule,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight({10_000})]
+		pub fn set_organizer(origin: OriginFor<T>, organizer: T::AccountId) -> DispatchResult {
+			ensure_root(origin)?;
+
+			Organizer::<T>::put(&organizer);
+			Self::deposit_event(Event::OrganizerSet { organizer });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight({10_000})]
+		pub fn add_rule_to(
+			origin: OriginFor<T>,
+			extrinsic_id: ExtrinsicId,
+			rule: u8,
+		) -> DispatchResult {
+			let _ = Self::ensure_organizer(origin)?;
+
+			ensure!(
+				!AffiliateRules::<T>::contains_key(extrinsic_id),
+				Error::<T>::ExtrinsicAlreadyHasRule
+			);
+			AffiliateRules::<T>::insert(extrinsic_id, rule);
+			Self::deposit_event(Event::RuleAdded { extrinsic_id });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight({10_000})]
+		pub fn clear_rule_from(origin: OriginFor<T>, extrinsic_id: ExtrinsicId) -> DispatchResult {
+			let _ = Self::ensure_organizer(origin)?;
+
+			AffiliateRules::<T>::remove(extrinsic_id);
+			Self::deposit_event(Event::RuleCleared { extrinsic_id });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight({10_000})]
+		pub fn affiliate_to(origin: OriginFor<T>, affiliator: T::AccountId) -> DispatchResult {
+			let affiliatee = ensure_signed(origin)?;
+
+			ensure!(affiliator != affiliatee, Error::<T>::CannotAffiliateSelf);
+			ensure!(
+				!Affiliators::<T>::contains_key(&affiliatee),
+				Error::<T>::CannotAffiliateExistingAffiliator
+			);
+			ensure!(
+				!Affiliatees::<T>::contains_key(&affiliatee),
+				Error::<T>::CannotAffiliateAlreadyAffiliatedAccount
+			);
+			ensure!(
+				Affiliatables::<T>::contains_key(&affiliator),
+				Error::<T>::CannotAffiliateToAccount
+			);
+
+			Self::add_new_affiliate_to(affiliator.clone(), affiliatee.clone())?;
+
+			Self::deposit_event(Event::AccountAffiliated { account: affiliatee, to: affiliator });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight({10_000})]
+		pub fn clear_affiliation(origin: OriginFor<T>) -> DispatchResult {
+			let affiliatee = ensure_signed(origin)?;
+
```

Comment:
We should add comment, why we don't change anything else then removing the linked accounts.